test/unit/app/common/lib/menuUtilTest.js

```diff
@@ -224,11 +228,44 @@ describe('menuUtil', function () {
         { title: 'site10', location: 'https://brave10.com' },
         { title: 'site11', location: 'https://brave11.com' }
       ])
-      const menuItems = menuUtil.createRecentlyClosedMenuItems(windowStateClosedFrames)
+      const menuItems = menuUtil.createRecentlyClosedTemplateItems(windowStateClosedFrames)
 
       assert.equal(menuItems.length, 12)
       assert.equal(menuItems[2].label, windowStateClosedFrames.get(1).get('title'))
       assert.equal(menuItems[11].label, windowStateClosedFrames.get(10).get('title'))
     })
   })
+
+  describe('sanitizeTemplateItems', function () {
+    it('removes entries which are falsey', function () {
+      const template = [null, undefined, false, {label: 'lol'}]
+      const result = menuUtil.sanitizeTemplateItems(template)
+      const expectedResult = [{label: 'lol'}]
+      assert.deepEqual(result, expectedResult)
+    })
+    it('removes duplicate menu separators', function () {
+      const template = [separator, separator, {label: 'lol'}]
+      const result = menuUtil.sanitizeTemplateItems(template)
+      const expectedResult = [separator, {label: 'lol'}]
+      assert.deepEqual(result, expectedResult)
+    })
+    it('removes items which are missing label or type', function () {
+      const template = [{}, {test: 'test'}, {label: 'lol'}]
+      const result = menuUtil.sanitizeTemplateItems(template)
+      const expectedResult = [{label: 'lol'}]
+      assert.deepEqual(result, expectedResult)
+    })
+    it('removes items which have non-string values for label or type', function () {
+      const template = [{label: true}, {type: function () { console.log('test') }}, {label: 'lol'}]
+      const result = menuUtil.sanitizeTemplateItems(template)
+      const expectedResult = [{label: 'lol'}]
+      assert.deepEqual(result, expectedResult)
+    })
+    it('always returns an array (even for one item)', function () {
```

Comment:
This last test here is the one I added which failed (revealing the bug). The Array.reduce that I was doing would returning a non-array when only 1 entry exists (it now properly returns an array)